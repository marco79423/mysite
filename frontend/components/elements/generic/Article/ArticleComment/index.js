import ArticleComment from './ArticleComment'

export default ArticleComment
