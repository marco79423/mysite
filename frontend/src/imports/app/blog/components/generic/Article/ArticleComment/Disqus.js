import React from 'react'
import PropTypes from 'prop-types'

export default class Disqus extends React.Component {
  static PropTypes = {
    /**
     * ref: https://github.com/mzabriskie/react-disqus-thread/blob/master/lib/components/DisqusThread.js
     * ref: https://github.com/disqus/DISQUS-API-Recipes/blob/master/snippets/js/disqus-reset/disqus_reset.html
     */
    shortname: PropTypes.string.isRequired,
  }

  componentDidMount () {
    this.loadDisqus()
  }

  componentDidUpdate () {
    this.loadDisqus()
  }

  loadDisqus = () => {
    // If Disqus has already been added, reset it
    if (typeof DISQUS !== 'undefined') {
      DISQUS.reset({
        reload: true,
        config: function config () {
          this.page.url = this.page.url.replace(/#/, '') + '#!newthread'
        }
      })
    } else { // Otherwise add Disqus to the page
      const child = document.createElement('script')
      child.async = true
      child.type = 'text/javascript'
      child.src = '//' + this.props.shortname + '.disqus.com/embed.js'

      const parent = document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]
      parent.appendChild(child)
    }
  }

  render () {
    return (
      <div>
        <div id="disqus_thread"/>
        <a href="http://disqus.com" className="dsq-brlink">
          Blog comments powered by <span className="logo-disqus">Disqus</span>.
        </a>
      </div>
    )
  }
}
