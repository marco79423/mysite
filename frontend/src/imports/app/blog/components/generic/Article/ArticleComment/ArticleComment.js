import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import Disqus from './Disqus'

export default class ArticleComment extends React.PureComponent {
  static PropTypes = {
    config: ImmutablePropTypes.map
  }

  render () {
    if (!this.props.config) {
      return <div/>
    }
    return (
      <Disqus shortname={this.props.config.get('shortName')}/>
    )
  }
}
