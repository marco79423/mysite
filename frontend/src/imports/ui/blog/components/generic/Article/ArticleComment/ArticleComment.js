import React from 'react'

import Disqus from './Disqus'

export default class ArticleComment extends React.PureComponent {

  render() {
    if (!this.props.config) {
      return <div/>
    }
    return (
      <Disqus shortname={this.props.config.shortName}/>
    )
  }
}
