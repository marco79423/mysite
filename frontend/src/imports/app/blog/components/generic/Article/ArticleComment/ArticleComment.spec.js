import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import ArticleComment from './ArticleComment'

test('It should render ArticleComment component correctly', () => {
  const config = Immutable.Map({
    shortName: 'marco79423'
  })
  const component = renderer.create(
    <ArticleComment config={config}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('It should not crash when config is null', () => {
  const component = renderer.create(
    <ArticleComment config={null}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
