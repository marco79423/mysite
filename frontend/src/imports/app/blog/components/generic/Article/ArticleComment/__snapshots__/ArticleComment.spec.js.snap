// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should not crash when config is null 1`] = `<div />`;

exports[`It should render ArticleComment component correctly 1`] = `
<div>
  <div
    id="disqus_thread"
  />
  <a
    className="dsq-brlink"
    href="http://disqus.com"
  >
    Blog comments powered by 
    <span
      className="logo-disqus"
    >
      Disqus
    </span>
    .
  </a>
</div>
`;
